src/utils/hooks/transaction/hooks/use-fee-estimate.ts

```diff
@@ -0,0 +1,200 @@
+import { CurrencyExt, isCurrencyEqual } from '@interlay/interbtc-api';
+import { MonetaryAmount } from '@interlay/monetary-js';
+import { mergeProps } from '@react-aria/utils';
+import { Key, useCallback, useRef, useState } from 'react';
+import { useErrorHandler } from 'react-error-boundary';
+import { MutationFunction, useMutation, UseMutationOptions, UseMutationResult } from 'react-query';
+import { useSelector } from 'react-redux';
+import { useInterval } from 'react-use';
+
+import { StoreType } from '@/common/types/util.types';
+import { SelectProps } from '@/component-library';
+import { GOVERNANCE_TOKEN } from '@/config/relay-chains';
+import { REFETCH_INTERVAL } from '@/utils/constants/api';
+
+import { useGetLiquidityPools } from '../../api/amm/use-get-liquidity-pools';
+import { useGetBalances } from '../../api/tokens/use-get-balances';
+import { useGetCurrencies } from '../../api/use-get-currencies';
+import { Actions, Transaction, TransactionArgs } from '../types';
+import { estimateTransactionFee, getActionAmount } from '../utils/fee';
+import { getActionData } from '../utils/params';
+
+// we are deducting the fee from the action amount when the user applies max,
+// so we only need to check if the balance can atleast cover for the fee amount.
+const isFeeValid = (
+  amount: MonetaryAmount<CurrencyExt>,
+  balance?: MonetaryAmount<CurrencyExt>,
+  actionAmount?: MonetaryAmount<CurrencyExt>
+): boolean => {
+  if (!balance) {
+    return true;
+  }
+
+  // when there isn't action amount involved, we check for greater or equal
+  if (!actionAmount) {
+    return balance.gte(amount);
+  }
+
+  const isActionAmountValid = actionAmount.gt(balance);
+
+  if (!isActionAmountValid) {
```

Comment:
I think this naming would make more sense - the actoin amount is invalid if it's greater than account's balance, no?
```suggestion
  const isActionAmountInvalid = actionAmount.gt(balance);

  if (isActionAmountInvalid) {
```